### PR TITLE
Treewide: Add new profile for Apple MacBook Pro 8,1

### PR DESCRIPTION
**Pull request by nakibrayan3**
###### Description of changes

I added a new profile for Apple MacBook Pro 8,1

###### Things done

- [x] Tested the changes in your own NixOS Configuration
- [x] Tested the changes end-to-end by using your fork of `nixos-hardware` and
      importing it via Flake input

### CODE REVIEW

File: apple/macbook-pro/8-1/default.nix [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  imports = [
+    ../.
+    ../../../common/gpu/intel/sandy-bridge
+    ../../../common/pc/laptop/ssd
+  ];
+
+  boot.initrd.availableKernelModules = [ "uhci_hcd" "ehci_pci" "ahci" "firewire_ohci" "usbhid" "usb_storage" "sd_mod" "sdhci_pci" ];
+  boot.kernelModules = [ "kvm-intel" ];
+
+  nixpkgs.config.allowUnfreePredicate = pkg: builtins.elem (lib.getName pkg) [ "b43-firmware" ];
```
Comment (Mic92) [inline apple/macbook-pro/8-1/default.nix:13]: Enable unfree is probably better left to the people in their own configuration.

```suggestion
```

File: apple/macbook-pro/8-1/default.nix [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+{
+  imports = [
+    ../.
+    ../../../common/gpu/intel/sandy-bridge
+    ../../../common/pc/laptop/ssd
+  ];
+
+  boot.initrd.availableKernelModules = [ "uhci_hcd" "ehci_pci" "ahci" "firewire_ohci" "usbhid" "usb_storage" "sd_mod" "sdhci_pci" ];
+  boot.kernelModules = [ "kvm-intel" ];
```
Comment (Mic92) [inline apple/macbook-pro/8-1/default.nix:11]: We don't replicate configuration from nixos-generate-config in nixos-hardware:

```suggestion
```

File: apple/macbook-pro/8-1/default.nix [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+  nixpkgs.config.allowUnfreePredicate = pkg: builtins.elem (lib.getName pkg) [ "b43-firmware" ];
+
+  networking.enableB43Firmware = lib.mkDefault true;
+
+  hardware.enableRedistributableFirmware = lib.mkDefault true;
+  hardware.cpu.intel.updateMicrocode = lib.mkDefault true;
```
Comment (Mic92) [inline apple/macbook-pro/8-1/default.nix:18]: Same as above:

```suggestion
```

File: apple/macbook-pro/8-1/default.nix [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  ];
+
+  boot.initrd.availableKernelModules = [ "uhci_hcd" "ehci_pci" "ahci" "firewire_ohci" "usbhid" "usb_storage" "sd_mod" "sdhci_pci" ];
+  boot.kernelModules = [ "kvm-intel" ];
+
+  nixpkgs.config.allowUnfreePredicate = pkg: builtins.elem (lib.getName pkg) [ "b43-firmware" ];
+
+  networking.enableB43Firmware = lib.mkDefault true;
+
+  hardware.enableRedistributableFirmware = lib.mkDefault true;
```
Comment (Mic92) [inline apple/macbook-pro/8-1/default.nix:17]: Afaik not-detected.nix should enable this (part of nixos-generate-config)

```suggestion
```

**Comment by nakibrayan3** [comment]
@Mic92 I applied all the suggestions you gave me.

**Comment by Mic92** [comment]
@mergify queue

**Pull request merged**